GitUI/GitModuleForm.cs

```diff
@@ -72,5 +84,30 @@ protected bool ExecuteScriptCommand(int command)
         {
             return Script.ScriptRunner.ExecuteScriptCommand(this, Module, command);
         }
+
+        protected bool checkDiffSelectedItemsLimit(List<GitItemStatus> list)
+        {
+            bool ret = true;
+
+            string diffWarnLimitStr = Module.GetEffectiveSetting("diff.warn.limit");
```

Comment:
You shouldn't introduce custom .gitconfig settings, but you can add an option to `GitCommands\Settings\AppSettings.cs`